tests/bgp/test_bgp_session_flap.py

```diff
@@ -0,0 +1,204 @@
+import time
+from tests.common.utilities import InterruptableThread
+import textfsm
+import traceback
+
+from natsort import natsorted
+
+logger = logging.getLogger(__name__)
+max_wait = 0
+wait_time = 2
+stop_threads = False
+proc_textfsm = "./bgp/templates/show_proc_cpu.template"
+bgp_sum_textfsm = "./bgp/templates/bgp_summary.template"
+flap_threads = []
+skip_hosts = []
+
+pytestmark = [
+    pytest.mark.topology('t1')
+]
+
+
+def get_cpu_stats(dut):
+    proc_cpu = dut.shell("show processes cpu | head -n 20", module_ignore_errors=True)['stdout']
+    proc_mem = dut.shell("show processes memory | head -n 20", module_ignore_errors=True)['stdout']
+    proc_sum = dut.shell("show processes summary | grep -v '0.0 0.0'", module_ignore_errors=True)['stdout']
+    bgp_cpu = dut.shell("show processes cpu | grep bgp", module_ignore_errors=True)['stdout']
+    bgp_v4_sum = dut.shell("show ip bgp summary | grep memory", module_ignore_errors=True)['stdout']
+    bgp_v6_sum = dut.shell("show ipv6 bgp summary | grep memory", module_ignore_errors=True)['stdout']
+    logger.info("CPU:\n{}\nMemory:\n{}\nSummary:\n{}\nBGP Memory:\n{}\nBGP IPv4:\n{}\nIPv6:\n{}\n"
+                .format(proc_cpu, proc_mem, proc_sum, bgp_cpu, bgp_v4_sum, bgp_v6_sum))
+    with open(proc_textfsm) as template:
+        fsm = textfsm.TextFSM(template)
+        parsed_cpu = fsm.ParseText(proc_cpu)[0]
+
+    with open(bgp_sum_textfsm) as template:
+        fsm = textfsm.TextFSM(template)
+        parsed_ipv4 = fsm.ParseText(bgp_v4_sum)[0]
+        parsed_ipv6 = fsm.ParseText(bgp_v6_sum)[0]
+    data = [float(parsed_cpu[0]), float(parsed_cpu[1]), float(parsed_cpu[2]),
+            float(parsed_ipv4[0]), float(parsed_ipv4[1]), float(parsed_ipv4[2]),
+            float(parsed_ipv6[0]), float(parsed_ipv6[1]), float(parsed_ipv6[2])]
+    return data
+
+
+@pytest.fixture(scope='module')
+def setup(tbinfo, nbrhosts, duthosts, rand_one_dut_hostname, enum_asic_index):
+    duthost = duthosts[rand_one_dut_hostname]
+
+    tor_neighbors = dict()
+    tor1 = natsorted([neighbor for neighbor in nbrhosts.keys() if neighbor.endswith('T0')])[0]
+
+    bgp_facts = duthost.bgp_facts(instance_id=enum_asic_index)['ansible_facts']
+    neigh_asn = dict()
+    for k, v in bgp_facts['bgp_neighbors'].items():
+        if v['description'] not in skip_hosts:
+            neigh_asn[v['description']] = v['remote AS']
+            tor_neighbors[v['description']] = nbrhosts[v['description']]["host"]
+            assert v['state'] == 'established'
+
+    # verify sessions are established
+    logger.info(duthost.shell('show ip bgp summary'))
+    logger.info(duthost.shell('show ipv6 bgp summary'))
+
+    setup_info = {
+        'duthost': duthost,
+        'neighhost': tor_neighbors[tor1],
+        'neigh_asn': neigh_asn[tor1],
+        'asn_dict':  neigh_asn,
+        'neighbors': tor_neighbors
+    }
+
+    logger.info("DUT BGP Config: {}".format(duthost.shell("show run bgp", module_ignore_errors=True)))
```

Comment:
can you change `show run bgp` to `vtysh -n <namespace> -c "show run bgp"` to support multi asic platform, please check the sonic-mgmt tests for references

Reply:
I made this change.